lib/pure/collections/tables.nim

```diff
@@ -240,6 +240,26 @@ iterator pairs*[A, B](t: Table[A, B]): (A, B) =
   for h in 0..high(t.data):
     if isFilled(t.data[h].hcode): yield (t.data[h].key, t.data[h].val)
 
+import algorithm
+
+iterator sortedPairs*[A, B](t: Table[A, B],
+  cmp: proc (x, y: A): int {.closure.};
```

Comment:
Tbh, i'm not even sure if its a good idea to expose this as a public proc. Such api can easily be misused in cases where OrderedTable would be more appropriate.